src/main/java/com/matera/account/account/AccountService.java

```diff
@@ -0,0 +1,56 @@
+package com.matera.account.account;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Service
+public class AccountService {
+
+    @Autowired
+    private AccountRepository accountRepository;
+
+    public List<Account> findAll() {
+        return accountRepository.findAll();
+    }
+
+    public List<Account> findClientAccountsByClientId(UUID clientId) {
+        List<Account> allAccounts = accountRepository.findAll();
+        return allAccounts.stream()
+                .filter(account -> account.getClientId().equals(clientId))
+                .collect(Collectors.toList());
+    }
+
+    public Account findAccountByAccountId(UUID clientId, UUID accountId) {
+        List<Account> allAccounts = findClientAccountsByClientId(clientId);
```

Comment:
same here, we want precise queries :)

adding something like this on your repository might do the trick

`@Query("SELECT a FROM Account a WHERE a.accountId = :accountId AND a.clientId = :clientId")
    public Optional<Account> findByAccountIdAndClientId(@Param("accountId") UUID accountId, @Param("clientId") UUID clientId);`

this query sintaxe is called JPQL, it's like querying the database, but using your entity instead :)